deep_qa/training/text_trainer.py

```diff
@@ -491,10 +472,12 @@ def _set_padding_lengths_from_model(self):
 
     ########################
     # Model-specific methods - if you do anything complicated, you probably need to override these,
-    # but simple models might be able to get by with just the default implementation
+    # but simple models might be able to get by with just the default implementation.  Some of
```

Comment:
`__func` is how you actually implement private methods in python (see [here](http://radek.io/2011/07/21/private-protected-and-public-in-python/), [here](http://stackoverflow.com/a/1641236/1467533), and [here](http://stackoverflow.com/questions/1641219/does-python-have-private-variables-in-classes#comment3435444_1641305)).  `_func` is "protected", by convention, which is why pylint complains about "protected-access" (_not_ "private-access") when you try to use them outside of the class where they were defined.

As we're building something that's actually going to be released, being really clear in the code and in the documentation about what's public, what's protected, and what's private is pretty important.  So, yes, I think we should definitely make use of python's conventions for all three.